### PR TITLE
libcurl: Fix missing dl link

### PR DESCRIPTION
**Pull request by SirLynix**
See #3118

* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

**Comment by SirLynix** [comment]
I don't get why it fails to find the Security frameworks on arm64 targets now, it worked fine here: https://github.com/xmake-io/xmake-repo/pull/3102

**Comment by SirLynix** [comment]
Probably because of https://github.com/xmake-io/xmake/commit/45f0750451087954ca4d649192d593341dd809f9?

**Comment by waruqi** [comment]
I think we should patch cmakelist.txt to add `dl` links. cmake build errors now.

**Comment by waruqi** [comment]
> Probably because of [xmake-io/xmake@45f0750](https://github.com/xmake-io/xmake/commit/45f0750451087954ca4d649192d593341dd809f9)?

maybe, I have improved it.

**Comment by waruqi** [comment]
> Probably because of [xmake-io/xmake@45f0750](https://github.com/xmake-io/xmake/commit/45f0750451087954ca4d649192d593341dd809f9)?

But if I revert it, it will find x86_64 libs on macOS/x86_64 when we build macos/arm64. `xmake f -a arm64` on macOS/x86_64

**Comment by SirLynix** [comment]
Is there a reason it can't find the Security framework on ARM64? I'm not familiar with macOS

**Comment by waruqi** [comment]
I don't know, I revert it first now.

File: packages/l/libcurl/xmake.lua [hunk 1]
```diff
@@ -16,7 +16,7 @@ package("libcurl")
     if is_plat("macosx", "iphoneos") then
         add_frameworks("Security", "CoreFoundation", "SystemConfiguration")
     elseif is_plat("linux") then
-        add_syslinks("pthread")
+        add_syslinks("pthread", "dl")
```
Comment (waruqi) [inline packages/l/libcurl/xmake.lua:19]: we should add dl to cmakelists.txt

Comment (SirLynix) [reply packages/l/libcurl/xmake.lua:19]: but why? #3118 shows an error when testing the package, so after cmake built libcurl successfully

Comment (waruqi) [reply packages/l/libcurl/xmake.lua:19]: not on_test, cmake is linking curl in on_install

> 100%] Linking C executable curl
cd /home/camel/.xmake/cache/packages/2401/l/libcurl/8.5.0/source/build_0a140039/src && /home/camel/.xmake/packages/c/cmake/3.28.1/130fb2d927cc4c0cabd31f0f6b3ef621/bin/cmake -E cmake_link_script CMakeFiles/curl.dir/link.txt --verbose=1

/usr/bin/cc -W -Wall -pedantic -Wbad-function-cast -Wconversion -Winline -Wmissing-declarations -Wmissing-prototypes -Wnested-externs -Wno-long-long -Wno-multichar -Wpointer-arith -Wshadow -Wsign-compare -Wundef -Wunused -Wwrite-strings -Waddress -Wattributes -Wcast-align -Wdeclaration-after-statement -Wdiv-by-zero -Wempty-body -Wendif-labels -Wfloat-equal -Wformat-security -Wignored-qualifiers -Wmissing-field-initializers -Wmissing-noreturn -Wno-format-nonliteral -Wno-sign-conversion -Wno-system-headers -Wredundant-decls -Wold-style-definition -Wstrict-prototypes -Wtype-limits -Wunreachable-code -Wunused-parameter -Wvla -Wclobbered -Wmissing-parameter-type -Wold-style-declaration -Wstrict-aliasing=3 -Wformat=2 -Warray-bounds=2 -ftree-vrp -Wduplicated-cond -Wnull-dereference -fdelete-null-pointer-checks -Wshift-negative-value -Wshift-overflow=2 -Walloc-zero -Wduplicated-branches -Wformat-overflow=2 -Wformat-truncation=2 -Wrestrict -Warith-conversion -Wdouble-promotion -Wenum-conversion -Wpragmas -Wunused-const-variable -O3 -DNDEBUG CMakeFiles/curl.dir/slist_wc.c.o CMakeFiles/curl.dir/tool_binmode.c.o CMakeFiles/curl.dir/tool_bname.c.o CMakeFiles/curl.dir/tool_cb_dbg.c.o CMakeFiles/curl.dir/tool_cb_hdr.c.o CMakeFiles/curl.dir/tool_cb_prg.c.o CMakeFiles/curl.dir/tool_cb_rea.c.o CMakeFiles/curl.dir/tool_cb_see.c.o CMakeFiles/curl.dir/tool_cb_wrt.c.o CMakeFiles/curl.dir/tool_cfgable.c.o CMakeFiles/curl.dir/tool_dirhie.c.o CMakeFiles/curl.dir/tool_doswin.c.o CMakeFiles/curl.dir/tool_easysrc.c.o CMakeFiles/curl.dir/tool_filetime.c.o CMakeFiles/curl.dir/tool_findfile.c.o CMakeFiles/curl.dir/tool_formparse.c.o CMakeFiles/curl.dir/tool_getparam.c.o CMakeFiles/curl.dir/tool_getpass.c.o CMakeFiles/curl.dir/tool_help.c.o CMakeFiles/curl.dir/tool_helpers.c.o CMakeFiles/curl.dir/tool_hugehelp.c.o CMakeFiles/curl.dir/tool_ipfs.c.o CMakeFiles/curl.dir/tool_libinfo.c.o CMakeFiles/curl.dir/tool_listhelp.c.o CMakeFiles/curl.dir/tool_main.c.o CMakeFiles/curl.dir/tool_msgs.c.o CMakeFiles/curl.dir/tool_operate.c.o CMakeFiles/curl.dir/tool_operhlp.c.o CMakeFiles/curl.dir/tool_paramhlp.c.o CMakeFiles/curl.dir/tool_parsecfg.c.o CMakeFiles/curl.dir/tool_progress.c.o CMakeFiles/curl.dir/tool_setopt.c.o CMakeFiles/curl.dir/tool_sleep.c.o CMakeFiles/curl.dir/tool_stderr.c.o CMakeFiles/curl.dir/tool_strdup.c.o CMakeFiles/curl.dir/tool_urlglob.c.o CMakeFiles/curl.dir/tool_util.c.o CMakeFiles/curl.dir/tool_vms.c.o CMakeFiles/curl.dir/tool_writeout.c.o CMakeFiles/curl.dir/tool_writeout_json.c.o CMakeFiles/curl.dir/tool_xattr.c.o CMakeFiles/curl.dir/var.c.o CMakeFiles/curl.dir//lib/base64.c.o CMakeFiles/curl.dir//lib/dynbuf.c.o -o curl ../lib/libcurl.a -lpthread /home/camel/.xmake/packages/o/openssl/1.1.1-w/6c51ab6278e2479b883dffafac69fdaf/lib/libssl.a /home/camel/.xmake/packages/o/openssl/1.1.1-w/6c51ab6278e2479b883dffafac69fdaf/lib/libcrypto.a -lpthread //这里缺少-ldl

Comment (SirLynix) [reply packages/l/libcurl/xmake.lua:19]: See https://github.com/curl/curl/issues/6165, maybe there's an option

Comment (SirLynix) [reply packages/l/libcurl/xmake.lua:19]: It's a missing dl syslinks from openssl package

**Comment by waruqi** [comment]
thanks!

**Pull request merged**